docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,557 @@
+
+* Cross-platform shortcut creation for the distributed application thanks to a complete `menuinst`
+  [^menuinst] rewrite
+* PKG installers on macOS can be customized, signed and notarized
+* `constructor` can deploy more than one conda environment at once now
+* Extra files can be added to the installation without having to use an extra conda package
+* Improvements for license collection and analysis
+* Small fixes for reported installation size, progress reports and other cosmetic details
+
+The resulting product now is able to install napari across operating systems with fully working
+shortcuts that respect the activation mechanism of `conda` environments to guarantee that all
+dependencies work properly. Having `constructor` distribute multiple environments at once allow us
+to split the installation like this:
+
+* A `base` environment with `conda`, `mamba` and `pip` to manage the other `conda` environments.
+* A `napari-X.Y.Z` environment (X.Y.Z being the version of the bundled release) with napari and
+  its dependencies.
+
+This separation allows us to:
+
+* Handle napari updates by simply creating a fresh environment with the new version (more on this
+  in Milestone 4).
+* Apply reparations in the environment without worrying that a faulty plugin install can render
+  the whole environment non-functional.
+* If necessary, adding support for "napari projects": a napari installation with a specific
+  combination of plugins. Useful in the event that a user wants to use plugins that cannot be
+  installed in the same environment due to conflicting dependencies (more details on this in
+  Milestone 3).
+
+The installer relies on conda-forge to obtain the needed packages. Pre-release installers
+can be built thanks to the nightlies available on the napari channel.
+
+### Milestone 3: Adding support for conda packages in the plugin manager
+
+napari has its own plugin manager, which so far relied on `pip` to install packages available on
+PyPI. To make it compatible with conda packaging, three key changes are needed:
+
+1.  The list of packages on conda-forge does not necessarily match the one coming from PyPI. Right
+    now, the plugins on conda-forge are a _subset_ of those on PyPI, but this might change in the
+    future if some napari plugins become available on conda-forge but not PyPI due to packaging
+    limitations (e.g. availability of dependencies). As a result, the plugin manager needs to source
+    the list of plugins from a repository-agnostic source: the napari hub API. It must be noted that
+    napari hub currently uses PyPI as the ground truth for the list of published plugins and the
+    available versions. This might need to change in the future if the scenario described in this
+    bullet point becomes a reality.
+2.  Once the napari hub API is feeding the list, the plugin manager should only list those available
+    on conda-forge, marking the packages only published on PyPI as unavailable for now. In the
```

Comment:
just want to reiterate some opposition to this.  we can discuss further.
I would prefer a combination of 
1) user warnings when installing from pip or wheel
2) specific package pins (i.e. make it impossible to modify numpy/qt or other binaries we determine to be too important/fragile)

Reply:
@tlambert03 I kinda disagree with you here. Note above that all this only applies to the bundle. Yes, for some pure-python plugins that don't depend on anything exotic, we could "just" install with pip, but those are also the packages that would be trivial to port to conda-forge anyway.

Reply:
i feel relatively strongly about this one... even just in the bundle.  So would prefer that ultimately this went to a broader vote where the options are laid out clearly (at which point I'm happy to be outvoted)